lib/active_merchant/billing/gateways/conekta.rb

```diff
@@ -173,22 +173,27 @@ def parse(body)
         JSON.parse(body)
       end
 
-      def headers(meta)
+      def headers(options)
         {
-          "Accept" => "application/vnd.conekta-v#{options[:version]}+json",
+          "Accept" => "application/vnd.conekta-v#{@options[:version]}+json",
           "Accept-Language" => "es",
-          "Authorization" => "Basic " + Base64.encode64("#{options[:key]}:"),
+          "Authorization" => "Basic " + Base64.encode64("#{@options[:key]}:"),
           "RaiseHtmlError" => "false",
           "Conekta-Client-User-Agent" => {"agent"=>"Conekta ActiveMerchantBindings/#{ActiveMerchant::VERSION}"}.to_json,
-          "X-Conekta-Client-User-Agent" => user_agent,
-          "X-Conekta-Client-User-Metadata" => meta.to_json
+          "X-Conekta-Client-User-Agent" => conekta_client_user_agent(options),
+          "X-Conekta-Client-User-Metadata" => @options.to_json
         }
       end
 
+      def conekta_client_user_agent(options)
+        return user_agent unless options[:application]
+        JSON.dump(JSON.parse(user_agent).merge!({application: options[:application]}))
```

Comment:
Welp that's the first time I've seen JSON in a header. Probably won't be the last.